addons/account/demo/account_demo.py

```diff
@@ -0,0 +1,352 @@
+# -*- coding: utf-8 -*-
+import logging
+import time
+from datetime import timedelta
+from dateutil.relativedelta import relativedelta
+import base64
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.misc import file_open
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountChartTemplate(models.Model):
+    _inherit = "account.chart.template"
+
+    @api.model
+    def _get_demo_data(self):
+        yield self._get_demo_data_move()
+        yield self._get_demo_data_statement()
+        yield self._get_demo_data_reconcile_model()
+        yield self._get_demo_data_attachment()
+        yield self._get_demo_data_mail_message()
+        yield self._get_demo_data_mail_activity()
```

Comment:
Why make this a generator instead of just return a list?

Reply:
Because we have to use the xml ids created at different point.
For instance `_get_demo_data_attachment` uses xml_id defined in `_get_demo_data_move`.

Reply:
Wow that *really* deserves a comment explaining that the methods implicitly depend on one another then.

Reply:
I mean... It is the same behavior as when we load data trough xml...

Reply:
So? Loading data through XML has its own know behaviour, that doesn't mean the expectations translate into unrelated Python function implicitly depending on one another through side-effects.

Reply:
it can't harm to have one more comment explaining how the demo data are created.

Reply:
Done